### PR TITLE
Add ability to enable dev shm streaming on eiger

### PR DESCRIPTION
**Pull request by DominicOram**
See https://github.com/DiamondLightSource/hyperion/issues/1372

### Instructions to reviewer on how to test:
1. Confirm tests pass

### Checks for reviewer
- [ ] Would the PR title make sense to a scientist on a set of release notes
- [ ] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)
- [ ] If changing the API for a pre-existing device, ensure that any beamlines using this device have updated their Bluesky plans accordingly

### CODE REVIEW

File: src/dodal/devices/eiger.py [hunk 1]
```diff
@@ -144,6 +144,9 @@ def stop(self, *args):
         self.disarm_detector()
         stop_status &= self.disable_roi_mode()
         stop_status.wait(self.GENERAL_STATUS_TIMEOUT)
+        # See https://github.com/DiamondLightSource/hyperion/issues/1395
+        LOGGER.info("Turning off Eiger dev/shm streaming")
```
Comment (olliesilvester) [inline src/dodal/devices/eiger.py:148]: Nit: If we have `enable_dev_shm` set to off, I think we will get a logging message during `change_dev_shm`, and in Hyperion when XRC finishes, and again here - all saying that dev shm is being disabled

**Review by olliesilvester — APPROVED** [review]
Minor nit, but looks good

File: src/dodal/devices/eiger.py (hunk 1 above)
Comment (DominicOram) [reply src/dodal/devices/eiger.py:148]: I think that given the consequence of leaving it on accidentally is very bad it's actually good to be very specific in the logs that we're turning it off even if that doesn't actually matter

Comment (olliesilvester) [reply src/dodal/devices/eiger.py:148]: yeah, makes sense

**Pull request merged**